public/app/features/trails/DataTrail.tsx

```diff
@@ -106,11 +150,46 @@ export class DataTrail extends SceneObjectBase<DataTrailState> {
   }
 
   protected _variableDependency = new VariableDependencyConfig(this, {
-    variableNames: [VAR_DATASOURCE],
+    variableNames: [VAR_DATASOURCE, VAR_OTEL_RESOURCES, VAR_OTEL_DEPLOYMENT_ENV, VAR_OTEL_JOIN_QUERY],
     onReferencedVariableValueChanged: async (variable: SceneVariable) => {
       const { name } = variable.state;
+
       if (name === VAR_DATASOURCE) {
         this.datasourceHelper.reset();
+
+        // fresh check for otel experience
+        this.checkDataSourceForOTelResources();
+        // clear filters on resetting the data source
+        const adhocVariable = sceneGraph.lookupVariable(VAR_FILTERS, this);
+        if (adhocVariable instanceof AdHocFiltersVariable) {
+          adhocVariable.setState({ filters: [] });
+        }
+      }
+
+      // update otel variables when changed
+      if (this.state.useOtelExperience && (name === VAR_OTEL_DEPLOYMENT_ENV || name === VAR_OTEL_RESOURCES)) {
+        // HERE WE UPDATE THE OTEL EXPERIENCE ENGINE
```

Comment:
Since it is in capital letters should it be something left over from the debugging?